test/fixtures/15-target-hostapps/applications.json

```diff
@@ -0,0 +1,28 @@
+{
+	"intel-nuc": {
+		"user": "admin",
+		"is_host": true,
+		"is_public": true,
+		"device_type": "intel-nuc",
+		"app_name": "intel-nuc"
+	},
+	"intel-nuc-esr": {
+		"user": "admin",
+		"is_host": true,
+		"is_public": true,
+		"device_type": "intel-nuc",
+		"app_name": "intel-nuc-esr"
+	},
```

Comment:
@dfunckt although ESR requires a balenaCloud subscription and plan upgrade, there's no reason to ignore that here in open-balena, right? i.e., folks could still migrate an ESR OS over to open-balena and that should be supported.

Reply:
I think the guarantees ESR releases offer are only relevant for the cloud by definition and are otherwise just balenaOS. In that sense, there's no reason for open-balena to not be compatible with ESR -- that doesn't imply anything. However, I'm not completely sure what you're asking or what this means technically as it seems the fact that a release is ESR is just semantics implied by the `-esr` suffix in the name. Can you clarify?

Reply:
I guess my question is really if we enforce the "no downgrades" policy in a hook here (like what we do for supervisors), ESR's different versioning scheme will effectively mean a device can never go from ESR -> non-ESR via supported methods. I think there's probably a use case for ESR -> non-ESR upgrades in open-balena, whereby the "no downgrades" policy must be enforced differently.

Reply:
I think I see. Going from ESR to regular balenaOS and vice versa is less about version downgrade/upgrade and more about migrating between different lines of software similar to, say, balenaOS for RPi and Rasbian -- they just happen to be both super compatible. I would thus encode all invariants into open-balena and think about providing the "downgrade" escape hatch separately -- it could be an action similar to changing the device type of an app.